components/tidb_query/src/util.rs

```diff
@@ -176,4 +194,17 @@ mod tests {
             ],
         );
     }
+
+    #[test]
+    fn test_get_rand() {
+        let mut rand = get_rng(None);
+        let res = (rand.borrow_mut()).gen::<f64>();
+        assert!(res < 1.0);
+        assert!(res >= 0.0);
+        let seed: u64 = 20191201;
+        rand = get_rng(Some(seed));
+        let res = (rand.borrow_mut()).gen::<f64>();
```

Comment:
Test whether the rand with seed results match TiDB. And you can then implement `rand_with_seed` push down without extra effort.

```SQL
TiDB> create table Test(id integer, title varchar(100));
Execute success (0.06 sec)
TiDB> insert into Test(id, title) values(1, "Hello");
Execute success (0.00 sec)
TiDB> insert into Test(id, title) values(2, "Hello2");
Execute success (0.00 sec)
TiDB> insert into Test(id, title) values(3, "Hello3");
Execute success (0.00 sec)
TiDB> select rand(1) from test;
+--------------------+
| rand(1)            | 
+--------------------+
| 0.6046602879796196 | 
| 0.9405090880450124 | 
| 0.6645600532184904 | 
+--------------------+
3 row in set (0.00 sec)
TiDB> select rand(2) from test;
+---------------------+
| rand(2)             | 
+---------------------+
| 0.16729663442585624 | 
| 0.2650543054337802  | 
| 0.05148802530284411 | 
+---------------------+
3 row in set (0.00 sec)
TiDB> select rand(3) from test;
+--------------------+
| rand(3)            | 
+--------------------+
| 0.7199826688373036 | 
| 0.6526308027999123 | 
| 0.9419605585830052 | 
+--------------------+
3 row in set (0.00 sec)
TiDB> select rand(id) from test;
+---------------------+
| rand(id)            | 
+---------------------+
| 0.6046602879796196  | 
| 0.16729663442585624 | 
| 0.7199826688373036  | 
+---------------------+
3 row in set (0.00 sec)
```

Reply:
Thank you very much.
I will implement rand_with_seed after this.